### PR TITLE
[common] Switch `int` to `index_t` for field indec variables in `pclBase`, add `pcl::UNAVAILABLE` for same

### PR DESCRIPTION
**Pull request by haritha-j**

### CODE REVIEW

File: common/include/pcl/pcl_base.h [hunk 1]
```diff
@@ -237,7 +240,7 @@ namespace pcl
       std::vector<int> field_sizes_;
```
Comment (kunaltyagi) [inline common/include/pcl/pcl_base.h:240]: Let's make another PR for this?

Comment (haritha-j) [reply common/include/pcl/pcl_base.h:240]: Since this is explicitly used for sizes, and thus cannot be negative, do you suppose `uindex_t` would be better for this?

Comment (kunaltyagi) [reply common/include/pcl/pcl_base.h:240]: I suppose so

Comment (haritha-j) [reply common/include/pcl/pcl_base.h:240]: I'll go ask the compilers.
